### PR TITLE
 Migrate VM List screen to PF4

### PR DESCRIPTION
**Pull request by rszwajko**
Depends on #1540    

Includes:
1. toolbars
2. tooltips
3. spinner
4. masthead icons
    
Fixes:
1. use translation independent filters
2. allow filter multiselection

### CODE REVIEW

**Comment by rszwajko** [comment]
### Overview with filters
image

### VM Actions on a card
image

### VM Actions inside toolbar
image
image

### Small screen mode
image
image
image

### Account Settings toolbar
image

### VM Console toolbar
image

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
Some tests failed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by rszwajko** [comment]
@sjd78 @sgratch 
Note that some styling issues has been fixed in https://github.com/oVirt/ovirt-web-ui/pull/1584 i.e. truncating action name on the VM card.

File: src/components/Toolbar/VmSort.js [hunk 1]
```diff
@@ -1,63 +1,63 @@
...
-            onClick={this.toggleCurrentSortDirection}
-          />
-        </Tooltip>
-      </Sort>
-    )
-  }
+      )}
+      isGrouped
+    />,
+    isAsc ? <SortAmountDownAltIcon key='altIcon'/> : <SortAmountDownIcon key='icon'/>,
```
Comment (sjd78) [inline src/components/Toolbar/VmSort.js:59]: Visuals are just a bit off with this icon.  A bit too small and no left margin from the sort by list.

Should it be clickable to toggle the asc/desc?

**Review by sjd78 — CHANGES_REQUESTED** [review]
At first the >4 cards in a row on larger widths was annoying, but it is growing on me.  The spacing between the name/status and action button should match the space between the icon and the name/status.  It looks a bit bigger. (My guess is `.vm-status` style doesn't need a `min-height`).

Your `VmActions` changes are ok with me, I just didn't look very hard.  The original code was a bit of a mess.  Yours looks to work well, so I'm just giving that bit a quick skim. `VmDetailsActions` and `VmDropdownActions` structure is :+1:.

VM List filter toolbar is looking :+1:!

Minimal tweaks suggested.

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by sjd78** [comment]
L&F issues noted (I'll resolve code comments containing them):

- On card ... The spacing between the name/status and action button should match the space between the icon and the name/status.  It looks a bit bigger. (My guess is `.vm-status` style doesn't need a `min-height`).

- Card Gallery ... There is something off with the gutters. The gallery container itself doesn't seem to have a left and right gutter / margin.  Looks like the scroll-container div has a top and bottom margin padding but needs a left and right to match.

- Toolbar ... `VmSort` font size and asc / desc icon left margin and size are off.  asc/desc icon clickable?

- Toolbar ... the "results of" text font size is smaller than expected

File: src/components/VmConsole/VmConsoleSelector.js [hunk 2]
```diff
@@ -2,15 +2,16 @@ import React, { useContext } from 'react'
...
 import { openConsole } from '_/actions'
-import style from './style.css'
+import SelectBox from '../SelectBox'
+import {
+  ToolbarItem,
+} from '@patternfly/react-core'
 
 const VmConsoleSelector = ({ vmId, vm, config, consoleType, isConsolePage, onOpenConsole, preferredConsole }) => {
-  const { msg } = useContext(MsgContext)
+  const { msg, locale } = useContext(MsgContext)
```
[the remark this answers is not in the document]
Comment (rszwajko) [reply src/components/VmConsole/VmConsoleSelector.js:14]: 1. we need to pass it down to `SelectBox` for sorting
2. this component is expected to be removed once we reach console upgrade PR 1585

**Comment by ovirt-infra** [comment]
All tests passed

File: src/components/Toolbar/VmsListToolbar.js [hunk 3]
```diff
@@ -1,123 +1,75 @@
...
-        <h5>
-          {
-            vms.get('filters').size
+    <>
+      <Toolbar className='vm-list-toolbar' clearAllFilters={onClearFilters}>
+        <ToolbarContent>
+          <VmFilters/>
+          <ToolbarItem>
+            <VmSort />
+          </ToolbarItem>
```
[the remark this answers is not in the document]
Comment (rszwajko) [reply src/components/Toolbar/VmsListToolbar.js]: This  looks better after removing PF3 styles in https://github.com/oVirt/ovirt-web-ui/pull/1584

**Review by sjd78 — APPROVED** [review]
LGTM

**Comment by sjd78** [comment]
OST fail:

```log
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {"method":"xpath","selector":"//div[@class='col-sm-12']/h5"}
```

**Comment by rszwajko** [comment]
Forced pushed with minor changes:
```

diff --git a/src/components/Toolbar/VmSort.js b/src/components/Toolbar/VmSort.js
index a4bd3cec..7f0d8cc1 100644
--- a/src/components/Toolbar/VmSort.js
+++ b/src/components/Toolbar/VmSort.js
@@ -61,8 +61,8 @@ const VmSort = ({ sort, msg, onSortChange }) => {
         />
       </ToolbarItem>
       <ToolbarItem>
-        <Button variant='plain' ariaLabel={msg.sortDirection()} onClick={() => onSortChange({ ...sort, isAsc: !isAsc })}>
-          {isAsc ? <SortAmountDownAltIcon/> : <SortAmountDownIcon key='icon'/>}
+        <Button variant='plain' aria-label={msg.sortDirection()} onClick={() => onSortChange({ ...sort, isAsc: !isAsc })}>
+          {isAsc ? <SortAmountDownAltIcon/> : <SortAmountDownIcon/>}
         </Button>
       </ToolbarItem>
     </ToolbarGroup>
diff --git a/src/components/Toolbar/VmsListToolbar.js b/src/components/Toolbar/VmsListToolbar.js
index e448d8e9..ac2989cc 100644
--- a/src/components/Toolbar/VmsListToolbar.js
+++ b/src/components/Toolbar/VmsListToolbar.js
@@ -33,6 +33,12 @@ const VmsListToolbar = ({ match, vms, pools, filters = {}, onClearFilters, msg }
           <VmFilters/>
           <VmSort />
           <ToolbarItem>
+            {
+            /* integration tests (OST) expect VM count under xpath //div[@class='col-sm-12']/h5
+            preserve this path:
+            keep the class 'col-sm-12' as marker class
+            keep the structure <div> followed by <h5> */
+            }
             <div className='col-sm-12'>
               <h5>
                 {

```

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by sjd78** [comment]
Most recent OST fail:
```
selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element &lt;a href="#" id="usermenu-logout"&gt;...&lt;/a&gt; is not clickable at point (1451, 100). Other element would receive the click: &lt;div class="pf-c-tooltip__content"&gt;...&lt;/div&gt;
```

**Pull request merged**